### PR TITLE
[struct_pack] support fast_varint encode, allow encode int as varint by config

### PR DESCRIPTION
**Pull request by poor-circle**
## Why

1. varint code is slow when serialization/deserialization. 
2. it's inconvenient to use struct_pack::var_int**_t, user may just want to use normal int type with varint encode.

## What is changing

1. add a new encode named fast_varint. It 's very fast when serialization/deserialization. It can be used if your int field is likely to be zero or it max value in stuct is small likely.
3. allow user encode normal int as varint to simply usage.

For encoding detail, see document[TODO]

## Example

Here is a struct:
```cpp
using namespace struct_pack;
struct example0 {
  int32_t a;
  int64_t b;
  uint32_t c;
  uint64_t d;
};
```
you can add config to change coding:
```cpp
struct example1 {
  int32_t a;
  int64_t b;
  uint32_t c;
  uint64_t d;
  static constexpr auto struct_pack_config =
      USE_FAST_VARINT | ENCODING_WITH_VARINT | DISABLE_ALL_META_INFO;
};
```
you can also change encoding config by ADL:
```cpp
struct example2 {
  int32_t a;
  int64_t b;
  uint32_t c;
  uint64_t d;
};

constexpr auto set_sp_config(example2*) {
  return USE_FAST_VARINT | ENCODING_WITH_VARINT | DISABLE_ALL_META_INFO;
}
```

### CODE REVIEW

File: include/ylt/struct_pack/packer.hpp [hunk 1]
```diff
@@ -143,11 +148,93 @@ class packer {
...
+          (serialize_one_fast_varint<parent_tag, 1>(items), ...);
+          break;
+        case 1:
+          (serialize_one_fast_varint<parent_tag, 2>(items), ...);
+          break;
+        case 2:
+          (serialize_one_fast_varint<parent_tag, 4>(items), ...);
+          break;
+        case 3:
+          (serialize_one_fast_varint<parent_tag, 8>(items), ...);
```
Comment (qicosmos) [inline include/ylt/struct_pack/packer.hpp:224]: check if has int64

File: include/ylt/struct_pack/unpacker.hpp [hunk 2]
```diff
@@ -596,12 +648,111 @@ class unpacker {
...
-  template <size_t size_type, uint64_t version, bool NotSkip, typename T>
+  template <uint64_t parent_tag, bool no_skip, std::size_t width,
+            std::size_t bitset_width, typename Arg>
+  constexpr struct_pack::errc STRUCT_PACK_INLINE deserialize_one_fast_varint(
+      std::bitset<bitset_width> &vec, int &i, Arg &item) {
+    if constexpr (varint_t<Arg, parent_tag>) {
+      constexpr auto real_width = (std::min)(width, sizeof(Arg));
+      if (!vec[i++])
+        return {};
+      else if constexpr (!no_skip) {
```
Comment (qicosmos) [inline include/ylt/struct_pack/unpacker.hpp:659]: remove else

File: include/ylt/struct_pack/unpacker.hpp [hunk 3]
```diff
@@ -596,12 +648,111 @@ class unpacker {
...
+        case 1:
+          ec = deserialize_fast_varint_helper<parent_tag, no_skip, 2>(vec, i,
+                                                                      items...);
+          break;
+        case 2:
+          ec = deserialize_fast_varint_helper<parent_tag, no_skip, 4>(vec, i,
+                                                                      items...);
+          break;
+        case 3:
+          ec = deserialize_fast_varint_helper<parent_tag, no_skip, 8>(vec, i,
```
Comment (qicosmos) [inline include/ylt/struct_pack/unpacker.hpp:739]: check if int64

**Comment by poor-circle** [comment]
I'll add document later.

**Pull request merged**